src/Pathy/Name.purs

```diff
@@ -24,32 +24,72 @@ derive newtype instance ordName :: Ord (Name a)
 instance showName :: Show (Name a) where
   show (Name name) = "(Name " <> show name <> ")"
 
--- | Retrieves the extension of a name.
+-- | Splits `Name` in name and extension part.
+-- |
+-- | ```purescript
+-- | splitName (Name ".foo")    == { name: ".foo", extension: Nothing }
+-- | splitName (Name "foo.")    == { name: "foo.", extension: Nothing }
+-- | splitName (Name "foo")     == { name: "foo",  extension: Nothing }
+-- | splitName (Name ".")       == { name: ".",    extension: Nothing }
```

Comment:
We can remove this example as it shuoldn't be actually used right? We can constrain this in the `IsName` class too, @garyb what you think?